Testing/Functional/Products/Tests/CallProductTests.ps1

```diff
@@ -7,23 +7,49 @@
 
 param(
     # The hashtable with the params.
+    [Parameter(Mandatory=$false)]
+    [ValidateNotNullOrEmpty()]
+    [boolean]$ReportErrors=$true,
+
     [Parameter(Mandatory)]
     [ValidateNotNullOrEmpty()]
-    [hashtable]$params,
+    [hashtable]$Params,
+
+
+    [Parameter(Mandatory=$false)]
+    [ValidateNotNullOrEmpty()]
+    [string]$Filter="*",
+
     # The thumbprint of the cert used to access the product.
     [Parameter(Mandatory)]
     [ValidateNotNullOrEmpty()]
-    [string]$thumbprint
+    [string]$Thumbprint
 )
 
-$testScriptDir = 'Testing/Functional/Products'
+$TestScriptDir = 'Testing/Functional/Products'
 
 # Add thumbprint to hashtable
-$params["Thumbprint"] = $thumbprint
+$Params["Thumbprint"] = $Thumbprint
 
 # Create an array of test containers
-$testContainers = @()
-$testContainers += New-PesterContainer -Path $testScriptDir -Data $params
+$TestContainers = @()
+$TestContainers += New-PesterContainer -Path $TestScriptDir -Data $Params
+
+$PesterConfig = @{
```

Comment:
I believe that CallProductTests is called by a lot of things.  Were you able to test *all* of these calls to make sure these changes didn't impact anything else?